h/static/scripts/group-forms/components/GroupModeration.tsx

```diff
@@ -1,11 +1,70 @@
+import { Spinner } from '@hypothesis/frontend-shared';
 import { useState } from 'preact/hooks';
 
 import type { Group } from '../config';
+import { useGroupAnnotations } from '../hooks/use-group-annotations';
+import type { APIAnnotationData } from '../utils/api';
 import GroupFormHeader from './GroupFormHeader';
 import type { ModerationStatus } from './ModerationStatusSelect';
 import ModerationStatusSelect from './ModerationStatusSelect';
 import FormContainer from './forms/FormContainer';
 
+type AnnotationsListProps = {
+  filterStatus?: ModerationStatus;
+};
+
+function AnnotationsList({ filterStatus }: AnnotationsListProps) {
+  const { loading, annotations } = useGroupAnnotations({ filterStatus });
+
+  return (
+    <div className="flex flex-col gap-y-2 mt-4">
+      <AnnotationsListContent
+        loading={loading}
+        annotations={annotations}
+        filterStatus={filterStatus}
+      />
+    </div>
+  );
+}
+
+type AnnotationsListContentProps = AnnotationsListProps & {
+  loading: boolean;
+  annotations: APIAnnotationData[];
+};
+
+function AnnotationsListContent({
+  loading,
+  annotations,
+  filterStatus,
+}: AnnotationsListContentProps) {
+  if (loading) {
+    return (
+      <div className="mx-auto">
+        <Spinner size="md" />
+      </div>
+    );
+  }
+
+  if (annotations.length === 0) {
+    return (
+      <div
+        className="border rounded p-2 text-center"
+        data-testid="annotations-fallback-message"
+      >
+        {filterStatus === 'PENDING'
+          ? 'You are all set!'
+          : 'No annotations found for selected status'}
+      </div>
+    );
+  }
+
+  return annotations.map(anno => (
+    <div key={anno.id} className="border rounded p-2">
+      {anno.text}
+    </div>
```

Comment:
We should add list / list item semantics for items in this list. In the client we use `<section>` for the container and `<article>` for individual items. I'm not sure that's optimal, but it would be consistent.